woocommerce/woocommerce-android
# Improve the handling of store switching after store creation

Closes: #10434


### Description
This PR improves our handling of the store switching after store creation, it adds two changes:
1. Logic to monitor the store creation if the store wasn't ready yet by the time the app was opened.
2. Show an alert before switching to the new store.

### Testing instructions
1. Open the app then create a store.
2. Pass the profiler steps
3. optionally: choose a theme.
4. In the store installation screen, close the app.
5. Re-open the app.
6. Wait a bit, then notice the dialog.
7. In Logcat, confirm the event: `site_creation_store_ready_alert_displayed`
8. Click on Switch Store.
9. Confirm the event: `site_creation_store_ready_alert_switch_store_tapped`
10. Confirm the app restarts.
11. If a theme was selected in 3, then a dialog will be shown to indicate them activation.

### Images/gif
https://github.com/woocommerce/woocommerce-android/assets/1657201/27211d3b-ea5b-48c3-92ec-bd8a7a9afb47
